File: kotlin-examples/src/main/java/java/ListOperations.java
package java;

import java.util.List;
import java.util.stream.Collectors;

public class ListOperations {
    public List<String> mapToOrderURLs(List<Integer> orderIds){
        List<String> refList = orderIds.stream()
                .map(id -> "v1/orders/"+id)
                .collect(Collectors.toList());
        return refList;
    }
}

File: kotlin-examples/src/main/java/kotlin/ListOperations.kt
package kotlin

// less boilerplate (stream(), collect()). result is already a list.
// string interpolation
fun mapToOrderURLs(orderIds: List<Int>): List<String> {
    val refs = orderIds.map { id -> "v1/orders/$id" }
    return refs
}
//or shorter: as single expression function and inferred return type for method
fun mapToOrderURLs2(orderIds: List<Int>) = orderIds.map { id -> "v1/orders/$id" }

//or even shorter: using "it" for the parameter (when there is only one parameter)
fun mapToOrderURLs3(orderIds: List<Int>) = orderIds.map { "v1/orders/$it" }